### DISCUSSION TITLE
Is it expected behaviour of juxt?

### DISCUSSION THREAD

**Discussion by yurkimus** [ramda/ramda#3435] [Q&A] [answered]
Hello, I was playing a bit with `juxt` utility in the official REPL ([code example](https://ramdajs.com/repl/?v=0.29.1#?const%20storageDelete%20%3D%20invoker%281%2C%20%27delete%27%29%0A%0Aconst%20storageEntries%20%3D%20compose%28Array.from%2C%20invoker%280%2C%20%27entries%27%29%29%0A%0Aconst%20getArgs1%20%3D%20juxt%28%5B%0A%20%20%28x%29%20%3D%3E%20storageDelete%28x%29%2C%0A%20%20storageEntries%0A%5D%29%0A%0Aconst%20getArgs2%20%3D%20juxt%28%5B%0A%20%20storageDelete%2C%0A%20%20storageEntries%0A%5D%29%0A%0AgetArgs1%28new%20URLSearchParams%28%7B%20a%3A%201%20%7D%29%29%20%2F%2F%20%3D%3E%20%5Bfunction%28t%29%7Breturn%20s.apply%28this%2Carguments%29%7D%2C%20%5B%5B%22a%22%2C%20%221%22%5D%5D%5D%0A%0AgetArgs2%28new%20URLSearchParams%28%7B%20a%3A%201%20%7D%29%29%20%2F%2F%20%3D%3E%20function%28t%29%7Breturn%20s.apply%28this%2Carguments%29%7D)) and found an issue.

The `getArgs1` function is returning array as expected, but the `getArgs2` doesn't.

Code:
```javascript
import { compose, invoker, juxt } from 'ramda'

const storageDelete = invoker(1, 'delete')

const storageEntries = compose(Array.from, invoker(0, 'entries'))

const getArgs1 = juxt([(x) => storageDelete(x), storageEntries])

const getArgs2 = juxt([storageDelete, storageEntries])

console.log(
  'getArgs1',
  getArgs1(new URLSearchParams({ a: 1 }))
) // => [ [Function (anonymous)], [ [ 'a', '1' ] ] ]

console.log(
  'getArgs2',
  getArgs2(new URLSearchParams({ a: 1 }))
) // => getArgs2 [Function (anonymous)]

```

If this is expected behaviour, can someone explain the semantics/reasons of it?

**Comment by jlissner** [comment, marked as answer]
Yes, this is expected, `storageDelete` has an arity of `2`, so `juxt` is waiting for the second argument before passing arguments, you can fix this by doing something like:

```js
const storageDelete = unary(invoker(1, 'delete'));

// or

const getArgs2 = juxt([unary(storageDelete), storageEntries])
```

Basically though, it's an artifact of ramda's auto-currying

**Comment by kedashoe** [comment]
ty for the answer @jlissner . Does it make sense @yurkimus ? `invoker` creates "n + 1" arity function, as it expects n arguments + the object. And `juxt` creates a function with the arity of maximum arity of its arguments.

Closing as this is expected, but let us know if you have any more questions

**Comment by yurkimus** [comment]
@jlissner, @kedashoe, thank you for the answer.

Should we make a note about this behaviour in the docs? Actual docs:

<img width="423" alt="image" src="https://github.com/ramda/ramda/assets/51003204/847ebd3d-1fca-4946-a041-ee1b074c5529">

I was trying to achieve just a list of functions in my example, not a new function based on maximum arity of its arguments. Looks not clear in the docs.

The `unary(func)` works well for that, thanks.

**Comment by kedashoe** [comment]
Sure, I think we'd happily accept a PR to add a note for this. While it is not documented for `juxt`, this is actually "standard" for ramda functions that take function arguments like this. For example [anyPass](https://ramdajs.com/docs/#anyPass) mentions it, as does [converge](https://ramdajs.com/docs/#converge) (which `juxt` uses under the hood).
